nixos/tests/systemd-coredump-disabled.nix

```diff
@@ -0,0 +1,19 @@
+import ./make-test-python.nix ({ pkgs, ... }:
+
+{
+  name = "systemd-coredump-disabled";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ squalus ];
+  };
+
+  nodes.machine = { pkgs, lib, ... }: {
+    systemd.coredump.enable = false;
+    systemd.package = pkgs.systemd.override {
+      withCoredump = false;
+    };
```

Comment:
might it make sense to add `withCoredump` to systemd passthru and default `systemd.coredump.enable` to `systemd.package.withCoredump`?

Reply:
I assume the PR intent is mostly to not fail ungracefully in case `systemd.package` is overridden with a variant that doesn't include coredump functionality.

Peeking at `systemd.package` passthru to set options sounds like the wrong direction.

Reply:
Agreed with @flotkli. I would rather the decision of how to handle coredumps was separate from whether systemd-coredump was built.